recipes/protobuf/all/conanfile.py

```diff
@@ -64,23 +81,37 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
+            # shared build protobuf requires Abseil as shared libraries to avoid ODR violations.
+            if Version(self.version) >= "4.22.0" and is_msvc(self):
+                self.options["abseil"].shared = True
```

Comment:
ODR violation of what?

Reply:
![image](https://github.com/conan-io/conan-center-index/assets/89922763/b90adb9c-094c-459b-9669-4088bf835ee5)

https://github.com/abseil/abseil-cpp/issues/125

Reply:
From what I read there, that is a generic ODR warning that applies to many libraries.

AFAIK, Conan is supposed to ensure all copies of Abseil (in this case) would be compiled the same way,
so it is safe to link Abseil static, as long as you don't link Abseil into multiple shared libraries as static.

So eg if you have LibA.dll and LibB.dll and both use Abseil, then you should link Abseil as shared.
That also applies if LibB.dll was program.exe (that linked LibA.dll)

But if only one dll used Abseil, or only the program.exe linked Abseil, then it would be safe to use Abseil as static.

Note that if you have LibA.lib and LibB.lib (both static libs) using Abseil, it is safe to link Abseil static too.  They will all get linked together in the final program.exe

Last note, if you DO have 2 dlls that link abseil statically, then as long as they both only use it internally, then it may be safe - both will use their own copies independently.

There are advantages to using static libs, you should use them where possible.
* easier to deploy (no extra dlls!)
* compiler optimiser has more chances to strip and inline things, especially if you use LTCG (link time code generation)
* no runtime linking and function call stub overheads etc

Reply:
@toge  is there any way we can fix that and get stuff merged? This blocks the grpc update, since it requires fresh protobuf and both these packages are extremely old and CVE'ed in conan...